### PR TITLE
Password check fixes

### LINKED ISSUE #897: Password "ASDF1234" gives: Password must contain at least one number or symbol.
While "asdf1234" works fine. Notice the case.

### PR DESCRIPTION
**Pull request by mikemurray**
resolves #897 

At the moment we only need to validate a password based on length. The
regex is unnecessary and redundant since we accept all characters.

Bonus: Cleanup and fix eslint issues in this file.

### CODE REVIEW

**Comment by mikemurray** [comment]
I blocked this because tests need to be written or updated to verify functionality

File: packages/reaction-accounts/tests/jasmine/server/integration/validation.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+
+describe("Account Registration Validation ", function () {
+  describe("usernamne valiation ", function () {
```
Comment (aaronjudd) [inline packages/reaction-accounts/tests/jasmine/server/integration/validation.js]: um, you mean:  `username validation`?

File: packages/reaction-accounts/tests/jasmine/server/integration/validation.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        const username = "tenten";
+        Meteor.call("accounts/validation/username", username, function (error, result) {
+          expect(result).toEqual(true);
+          return done();
+        });
+      }
+    );
+  });
+
+  describe("email address valiation ", function () {
```
Comment (aaronjudd) [inline packages/reaction-accounts/tests/jasmine/server/integration/validation.js]: `valiation` sp?

File: packages/reaction-accounts/tests/jasmine/server/integration/validation.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+          expect(result).toEqual(jasmine.any(Object));
+          return done();
+        });
+      }
+    );
+
+    it(
+      "should allow a valid email address",
+      done => {
+        const email = "email@website.com";
```
Comment (aaronjudd) [inline packages/reaction-accounts/tests/jasmine/server/integration/validation.js]: could use faker.internet.email in the future..

File: packages/reaction-accounts/client/helpers/validation.js [hunk 4]
```diff
@@ -11,76 +16,96 @@ window.LoginFormValidation = {
...
-    var validPasswordRegex = /^.*(?=.*[a-z])(?=.*[\d\W]).*$/;
-    options = options || {};
+  /**
+   * Password validation
+   * Passwords may be validated 2 ways.
+   * "exists" (options.validationLevel = "exists") - Password must not be blank. Thats is the only rule. Used to validate a sign in.
+   * undefined (options.validationLevel = undefined) - Password must meet the lenght and other criteria to validate. Used for validating a new sign up.
+   * @param  {String} password Password to validate
+   * @param  {Object} options Options to apply to the password validator
+   * @param  {String} options.validationLevel "exists" | undefined (default)
```
Comment (zenweasel) [inline packages/reaction-accounts/client/helpers/validation.js]: Is this actually a param, or just an important variable?

File: packages/reaction-accounts/tests/jasmine/server/integration/validation.js (hunk 1 above)
Comment (zenweasel) [inline packages/reaction-accounts/tests/jasmine/server/integration/validation.js]: Big +1 for all these tests

File: packages/reaction-i18n/private/data/i18n/en.json [hunk 5]
```diff
@@ -523,6 +523,7 @@
           "passwordRequired": "Password is required.",
           "passwordMustBeAtLeast6CharactersLong": "Password must be at least 6 characters long.",
           "passwordMustContainRequirements": "Password must contain at least one number or symbol.",
+          "usernameMinLength": "Username must be at least 3 characters long",
           "minChar": "6 character minimum password.",
           "pwdsDontMatch": "Passwords don't match",
           "pwOneDigit": "Password must have at least one digit.",
```
Comment (zenweasel) [inline packages/reaction-i18n/private/data/i18n/en.json]: (not your code but..) Don't these message contradict each other? Is it one number or symbol or one digit? Also we should probably rephrase "6 character minimum password" as "Password much contain at least 6 characters"

**Pull request merged**